js/post/update_post.js

```diff
@@ -5,200 +5,165 @@ var completed_spec = {'CSC108': 0, 'CSC148': 0, 'CSC165': 0, 'Calc1': 0, 'CSC207
-        }
-    }
 
     // Update 300s
-    var i = 0; 
-    var spec300s = $('.lvl300spec');
-    for (var m = 0; m < 3; m++) {
-        if (spec300s[m].value != '') {
-            i += 1;
-        }
-    }
-    if (i === 3) {
+    if (filled_textboxes_300['spec'] === 3) {
         updateCategory($('#spec_300')[0].getElementsByClassName('code')[0], 'fulfilled');
-        updateCategory($('#maj_300')[0].getElementsByClassName('code')[0], 'fulfilled');
         updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'fulfilled');
-
+        updateCategory($('#maj_300')[0].getElementsByClassName('code')[0], 'fulfilled');
+        categories_completed['spec'] += 1;
+        categories_completed['maj'] += 1;
     } else {
         updateCategory($('#spec_300')[0].getElementsByClassName('code')[0], 'not fulfilled');
         updateCategory($('#maj_300')[0].getElementsByClassName('code')[0], 'not fulfilled');
         updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'not fulfilled');
-        
     }
 
-    // Update 400s
-
-    i = 0; 
-    var spec400s = $('.lvl400spec');
-    for (var l = 0; l < 3; l++) {
-        if (spec400s[l].value != '') {
-            i += 1;
-        }
-    }
 
-    if (i === 3) {
+    // Update 400s
+    if (filled_textboxes_400['spec'] === 3) {
         updateCategory($('#spec_400')[0].getElementsByClassName('code')[0], 'fulfilled');
-        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'fulfilled');
         updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'fulfilled');
+        categories_completed['spec'] += 1;
     } else {
         updateCategory($('#spec_400')[0].getElementsByClassName('code')[0], 'not fulfilled');
-        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'not fulfilled');
         updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'not fulfilled');
     }
 
-    // Update Extra
-
-    countMaj = 0; 
-    countSpec = 0;
-    var specExtra = $('#spec_extra')[0].getElementsByTagName('input');
-    var majExtra = $('#maj_extra')[0].getElementsByTagName('input');
-    for (var l = 0; l < 4; l++) {
-        if (specExtra[l].value != '') {
-            countSpec += 1;
-        } if (l < 3) {
-            if (majExtra[l].value != '') {
-                countMaj += 1;
-            }
-        }
+    if (filled_textboxes_400['maj'] === 1) {
+        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'fulfilled');
+        categories_completed['maj'] += 1;
+    } else {    
+        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'not fulfilled'); 
     }
 
-    if (countSpec === 4) {
+    // Update Extra
+    if (filled_textboxes_extra['spec'] === 4) {
         updateCategory($('#spec_extra')[0].getElementsByClassName('code')[0], 'fulfilled');
-    } if (countMaj === 3) {
+        categories_completed['spec'] += 1;
+    } if (filled_textboxes_extra['maj'] === 3) {
```

Comment:
I could put each if statement on its own line instead of right beside the ending bracket to the last if statement. If that's not what you meant, please let me know! They need to be separate if statements because each POSt has a different number of textboxes to be filled to fulfill each category and each one needs to be checked.